### PR TITLE
Implement GPG related api

### PR DESCRIPTION
**Pull request by sapk**
In order to implement go-gitea/gitea#425

I added all the api call that should be implemented and struct needed to format requests and responses.
 
I follow the github format as a example: https://developer.github.com/v3/users/gpg_keys/

### CODE REVIEW

File: gitea/user_gpgkey.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// Copyright 2015 The Gogs Authors. All rights reserved.
```
Comment (lunny) [inline gitea/user_gpgkey.go]: error comment head `2017 Gitea`

**Review by lunny — CHANGES_REQUESTED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

File: gitea/user_gpgkey.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	CanEncryptStorage bool           `json:"can_encrypt_storage"`
+	CanCertify        bool           `json:"can_certify"`
+	Created           time.Time      `json:"created_at,omitempty"`
+	Expires           time.Time      `json:"expires_at,omitempty"`
+}
+
+// GPGKeyEmail a email attache to a GPGKey
+type GPGKeyEmail struct {
+	Email string `json:"email"`
+	//Verified bool   `json:"verified"`
```
Comment (Bwko) [inline gitea/user_gpgkey.go]: Nitpicking: Commented code?

Comment (sapk) [reply gitea/user_gpgkey.go]: It's in the github api format but i don't know if in gitea we should/could present it. Maybe an other solution could be to uncomment it and add omitempty ?

**Comment by sapk** [comment]
As said on gitter, I hesitate to use PublicGPG... in place of GPG... since it will only contain the public part of the keys like the ssh keys. I don't know what you think of it.

**Comment by lunny** [comment]
Could we give an option to let user chose and we can use one as default?

**Comment by sapk** [comment]
@lunny You are saying to let the possibility to users to add a private gpg key to gitea ? From my point of view it would not be a good idea normally you limit the existence of your gpg keys only on trusted media. 

My question was just for the format of object and function name to be as https://github.com/go-gitea/go-sdk/blob/master/gitea/user_key.go#L15 since it's the same case (gitea only need the public key)

**Comment by Bwko** [comment]
I think that GPG is clear enough, since we don't have the private key

File: gitea/user_gpgkey.go (hunk 2 above)
Comment (Bwko) [reply gitea/user_gpgkey.go]: We could supply it, so you could just uncomment it

**Comment by sapk** [comment]
Ok perfect i will stick with this to continue in the next part (API and UI) in gitea.

**Comment by Bwko** [comment]
LGTM

**Comment by bkcsoft** [comment]
@sapk Is this done?

**Comment by bkcsoft** [comment]
It follows the [spec](https://developer.github.com/v3/users/gpg_keys/) so I'm merging it anyway 😂

**Pull request merged**